source/frontend/src/components/common/mapFSM/machineDefinition/mapMachine.ts

```diff
@@ -0,0 +1,299 @@
+      on: {
+        REQUEST_FLY_TO_LOCATION: {
+          actions: assign({
+            requestedFlyTo: (_, event: any) => ({
+              bounds: null,
+              location: event.latlng,
+            }),
+          }),
+          target: 'pendingFlyTo',
+        },
+        REQUEST_FLY_TO_BOUNDS: {
+          actions: assign({
+            requestedFlyTo: (_, event: any) => ({
+              bounds: event.bounds,
+              location: null,
+            }),
+          }),
+          target: 'pendingFlyTo',
+        },
+        REQUEST_FIT_BOUNDS: {
+          actions: assign((context: any) => {
+            if (context.draftLocations.length > 0) {
+              context.requestedFitBounds = latLngBounds(context.draftLocations);
+            } else {
+              context.requestedFitBounds = defaultBounds;
+            }
+          }),
+          target: 'pendingFitBounds',
+        },
+      },
+    },
+    pendingFlyTo: {
+      on: {
+        PROCESS_FLY_TO: {
+          actions: assign({
+            requestedFlyTo: () => ({
+              bounds: null,
+              location: null,
+            }),
+          }),
+          target: 'nothingPending',
+        },
+      },
+    },
+    pendingFitBounds: {
+      on: {
+        PROCESS_FIT_BOUNDS: {
+          actions: assign({
+            requestedFitBounds: () => defaultBounds,
+          }),
+          target: 'nothingPending',
+        },
+      },
+    },
+  },
+};
+
+const selectedFeatureStates = {
+  initial: 'idle',
+  states: {
+    idle: {
+      on: {
+        MAP_CLICK: {
+          actions: [
+            assign({
+              isLoading: () => true,
+              mapLocationSelected: (_, event: any) => event.latlng,
+              mapFeatureSelected: (_, event: any) => null,
+              mapLocationFeatureDataset: () => null,
+            }),
+          ],
+          target: 'loading',
+        },
+        MAP_MARKER_CLICK: {
+          actions: [
+            assign({
+              isLoading: () => true,
+              mapLocationSelected: (_, event: any) => null,
+              mapFeatureSelected: (_, event: any) => event.featureSelected,
+              mapLocationFeatureDataset: () => null,
+            }),
+          ],
+          target: 'loading',
+        },
+        CLOSE_POPUP: {
+          actions: [
+            assign({
+              mapLocationSelected: () => null,
+              mapFeatureSelected: () => null,
+              mapLocationFeatureDataset: () => null,
+            }),
+          ],
+        },
+      },
+    },
+    loading: {
+      invoke: {
+        src: 'loadLocationData',
+        onDone: {
+          target: 'idle',
+          actions: [
+            assign({
+              isLoading: () => false,
+              showPopup: () => true,
+              mapLocationFeatureDataset: (context, event: any) => event.data,
+            }),
+            // 'navigateToProperty', // TODO:uncomment this to show property information sidebar
```

Comment:
I see, lets make sure we do something with this otherwise this will just come back as a bug. Do we want to ask the BAs?

Reply:
Yes, I will get Dev's input once this gets merged. There are side effects to having it always open as it is now